moveit_simple/src/moveit_simple.cpp

```diff
@@ -775,6 +775,44 @@ bool Robot::getIK(const Eigen::Affine3d pose, std::vector<double> & joint_point,
 
 
 
+bool Robot::isNearSingular(const std::vector<double> & joint_point) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  double threshold = 1e-10; // How to determine this threshold?
```

Comment:
I would pick a robot config, say axis 4 and 6 within 3 degrees of being aligned and see what the threshold should be set to in order to catch this.